docarray/index/backends/weaviate.py

```diff
@@ -0,0 +1,806 @@
+
+    def _build_auth_credentials(self):
+        dbconfig = self._db_config
+
+        if dbconfig.auth_api_key:
+            return weaviate.auth.AuthApiKey(api_key=dbconfig.auth_api_key)
+        elif dbconfig.username and dbconfig.password:
+            return weaviate.auth.AuthClientPassword(
+                dbconfig.username, dbconfig.password, dbconfig.scopes
+            )
+        else:
+            return None
+
+    def _create_schema(self) -> None:
+        schema = {}
+
+        properties = []
+        column_infos = self._column_infos
+
+        for column_name, column_info in column_infos.items():
+            # in weaviate, we do not create a property for the doc's embeddings
+            if column_name == self.embedding_column:
+                continue
+            if column_info.db_type == 'blob':
+                self.bytes_columns.append(column_name)
+            if column_info.db_type == 'number[]':
+                self.nonembedding_array_columns.append(column_name)
+            prop = {
+                "name": column_name
+                if column_name != 'id'
+                else DOCUMENTID,  # in weaviate, id and _id is a reserved keyword
+                "dataType": [column_info.db_type],
+            }
+            properties.append(prop)
+
+        # TODO: What is the best way to specify other config that is part of schema?
+        # e.g. invertedIndexConfig, shardingConfig, moduleConfig, vectorIndexConfig
+        #       and configure replication
+        # we will update base on user feedback
+        schema["properties"] = properties
+        schema["class"] = self._db_config.index_name
+
+        # TODO: Use exists() instead of contains() when available
+        #       see https://github.com/weaviate/weaviate-python-client/issues/232
+        if self._client.schema.contains(schema):
+            logging.warning(
+                f"Found index {self._db_config.index_name} with schema {schema}. Will reuse existing schema."
+            )
+        else:
+            self._client.schema.create_class(schema)
+
+    @dataclass
+    class DBConfig(BaseDocIndex.DBConfig):
+        host: str = 'http://localhost:8080'
+        index_name: str = 'Document'
+        username: str = None
+        password: str = None
+        scopes: List[str] = field(default_factory=lambda: ["offline_access"])
+        auth_api_key: str = None
+        embedded_options: Optional[EmbeddedOptions] = None
+
+    @dataclass
+    class RuntimeConfig(BaseDocIndex.RuntimeConfig):
+        default_column_config: Dict[Any, Dict[str, Any]] = field(
+            default_factory=lambda: {
+                np.ndarray: {},
+                docarray.typing.ID: {},
+                'string': {},
+                'text': {},
+                'int': {},
+                'number': {},
+                'boolean': {},
+                'number[]': {},
+                'blob': {},
+            }
+        )
+
+        batch_config: Dict[str, Any] = field(
```

Comment:
so here instead of having a batch_config we could have all keys that are currently inside of batch_config, directly in the RuntimeConfig